finality-aleph/src/sync/handler/mod.rs

```diff
@@ -1790,4 +1890,35 @@ mod tests {
         assert!(handler.handle_internal_request(&headers[1].id()).unwrap());
         assert!(!handler.handle_internal_request(&headers[1].id()).unwrap());
     }
+
+    //TODO(A0-2984): remove this after legacy sync is excised
```

Comment:
we dont have to remove

Reply:
We do actually – after removing the other parts the component will no longer satisfy the requirements of this test, but it also won't have to, as the only imports will be happening from the forest, thus never producing overzealous imports.